### PR TITLE
SAAS-756 list of PMM instances

### PR DESCRIPTION
**Pull request by nicolalamacchia**
Tests will be added in SAAS-802

image
image
image

https://user-images.githubusercontent.com/2530868/146852749-46250520-6015-4a9d-b9d6-35d2c1e13da9.mov

### CODE REVIEW

File: src/assets/pmm-server-instances.svg [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<svg width="25" height="24" viewBox="0 0 25 24" fill="none" xmlns="http://www.w3.org/2000/svg">
+<path fill-rule="evenodd" clip-rule="evenodd" d="M0.5 19.46C0.5 18.908 0.947 18.497 1.5 18.497H10.521C11.072 18.497 11.521 18.908 11.521 19.46V21.497H0.5V19.46V19.46Z" stroke="currentColor" stroke-linecap="round" stroke-linejoin="round"/>
+<path fill-rule="evenodd" clip-rule="evenodd" d="M8.49951 16.4971H3.49951V13.4971H8.49951V16.4971Z" stroke="currentColor" stroke-linecap="round" stroke-linejoin="round"/>
+<path d="M1.5 18.4971V12.9601C1.5 12.1321 2.171 11.4971 3 11.4971H9C9.828 11.4971 10.5 12.1321 10.5 12.9601V18.4561" stroke="currentColor" stroke-linecap="round" stroke-linejoin="round"/>
```
Comment (fabio-silva) [inline src/assets/pmm-server-instances.svg:4]: can we just indent this? Our linter never complains on this one, but all other SVGs are indented

File: src/pages/GettingStarted/GettingStartedPmmSection/GettingStartedPmmSection.tsx [hunk 2]
```diff
@@ -1,15 +1,60 @@
...
+
+  return (
+    <GettingStartedSection
+      description={Messages.connectPMMDescription}
+      title={Messages.connectPMMTitle}
+      linkTo={Routes.instances}
+      linkText={Messages.viewInstances}
+      isTicked={!!inventory?.length}
+      loading={loading || showLoader}
+      disabled={!inventory?.length}
```
Comment (fabio-silva) [inline src/pages/GettingStarted/GettingStartedPmmSection/GettingStartedPmmSection.tsx:56]:
```suggestion
      disabled={!!inventory?.length}
```

File: src/pages/ManagePmmInstances/PmmInstanceList/PmmInstanceList.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        width: '35%',
+      },
+      {
+        Header: Messages.id,
+        accessor: 'id',
+        width: '35%',
+      },
+      {
+        Header: Messages.url,
+        accessor: ({ url }) => (
```
Comment (fabio-silva) [inline src/pages/ManagePmmInstances/PmmInstanceList/PmmInstanceList.tsx:26]: This should be `Cell: ({ value }) =>` instead, where `value` is automatically considered a `string`. I believe you must add `<Column<PmmInstance>[]>` on line 13 for this to work: `(): <Column<PmmInstance>[]>`

**Review by fabio-silva — CHANGES_REQUESTED** [review]

File: src/assets/pmm-server-instances.svg (hunk 1 above)
Comment (atymchuk) [reply src/assets/pmm-server-instances.svg:4]: Given that all other svgs are indented, I'd say we should.

**Review by atymchuk — CHANGES_REQUESTED** [review]

File: src/pages/GettingStarted/GettingStartedPmmSection/GettingStartedPmmSection.tsx (hunk 2 above)
Comment (nicolalamacchia) [reply src/pages/GettingStarted/GettingStartedPmmSection/GettingStartedPmmSection.tsx:56]: @fabio-silva that's actually the correct logic, it has to be disabled if the length is zero.

File: src/assets/pmm-server-instances.svg (hunk 1 above)
Comment (nicolalamacchia) [reply src/assets/pmm-server-instances.svg]: done

File: src/pages/ManagePmmInstances/PmmInstanceList/PmmInstanceList.tsx (hunk 3 above)
Comment (nicolalamacchia) [reply src/pages/ManagePmmInstances/PmmInstanceList/PmmInstanceList.tsx:26]: @fabio-silva why is that? `url` has the right type in `accessor`, furthermore TS is not complaining. Is there a reason for those changes?

Re `<Column<PmmInstance>[]>`, it's already specified on line 12. It's either that or on line 13, I prefer to type `useMemo` :)

Comment (fabio-silva) [reply src/pages/ManagePmmInstances/PmmInstanceList/PmmInstanceList.tsx:26]: It's here in the API: https://react-table.tanstack.com/docs/api/useTable#column-options 
- `accessor` : `The data returned by an accessor should be primitive and sortable.`;
- `Cell`: `Must return valid JSX`.

I.e. `accessor` tells the table how to reach the data field, while `Cell` dictates how to render it

**Review by tiagomotasantos — APPROVED** [review]

**Review by fabio-silva — DISMISSED** [review]

File: src/pages/ManagePmmInstances/PmmInstanceList/PmmInstanceList.tsx (hunk 3 above)
Comment (nicolalamacchia) [reply src/pages/ManagePmmInstances/PmmInstanceList/PmmInstanceList.tsx]: good point! done

**Review by fabio-silva — APPROVED** [review]

**Review by tiagomotasantos — APPROVED** [review]

**Pull request merged**